### DISCUSSION TITLE
Update MultiSlider.shinyInput()

### DISCUSSION THREAD

**Discussion by davide-zinghini** [Appsilon/shiny.blueprint#96] [Q&A]
Hello everyone. I'm using shiny.blueprint package and it is very helpful. 

I created on the UI a MultiSlider with 3 handles using the MultiSlider.shinyInput() function. If I need to update the handles value server-side, how can I do this? 

For other default Shiny inputs it is easy, I can just use updateSelectInput, updateNumericInput, etc.

Thank you,
Davide

**Comment by kamilzyla** [comment]
Hi @davide-zinghini and thanks for your interest in shiny.blueprint!

Most `.shinyInput()` components can be updated with [`shiny.react::updateReactInput()`](https://appsilon.github.io/shiny.react/reference/updateReactInput.html). However, `MultiSlider.shinyInput()` was a tricky component and it uses a custom implementation which doesn't support updates.

It's still possible to have a MultiSlider which can be updated on the server, albeit it's not that pretty. You'll need to use the underlying `MultiSlider` React component directly and keep its value in a reactive. Check out this example:

```r
library(shiny.blueprint)

defaultValue <- c(3, 7)

shinyApp(
  ui = Card(
    reactOutput("slider"),
    Button.shinyInput("reset", "Reset")
  ),
  server = function(input, output) {
    value <- reactiveVal(defaultValue)
    observe(value(input$value)) |> bindEvent(input$value)
    observe(value(defaultValue)) |> bindEvent(input$reset)

    output$slider <- renderReact({
      MultiSlider(
        onChange = setInput("value"),
        MultiSliderHandle(value = value()[1]),
        MultiSliderHandle(value = value()[2])
      )
    })
  }
)
```

**Reply by davide-zinghini** [reply]
Hi @kamilzyla , thank you for your reply. I needed something similar to [shiny.react::updateReactInput()](https://appsilon.github.io/shiny.react/reference/updateReactInput.html) because I have to initialise and create the MultiSlider UI-side and later, using an observeEvent(), update its values server-side. Even if I don't trigger the actions inside the observeEvent, the MultiSlider have to exist.
